xdsl/printer.py

```diff
@@ -259,12 +259,17 @@ def print_dictionary(
         print_value: Callable[[V], None],
         delimiter: str = ", ",
     ) -> None:
-        for i, (key, value) in enumerate(elems.items()):
-            if i:
-                self.print_string(delimiter)
-            print_key(key)
-            self.print_string("=")
-            print_value(value)
+        if not elems:
+            self.print("{}")
+        else:
+            self.print("{ ")
+            for i, (key, value) in enumerate(elems.items()):
+                if i:
+                    self.print_string(delimiter)
+                print_key(key)
+                self.print_string(" = ")
+                print_value(value)
+            self.print(" }")
```

Comment:
Just noticed this, I don't think it belongs in a PR tagged as `dialects`, since it's actually a core change. Why is it necessary?

Reply:
It's not "necessary", I could just put this logic in the printer for AttributesOp, but I thought it would make sense to use the print_dictionary function and I (maybe incorrectly) thought that since it was used nowhere else that it would be safe to modify/improve

Reply:
Would be happy to split this change off into a separate pr if wanted

Reply:
Hmm, that's weird that it's not used... I'm also surprised none of the tests fail... My first instinct is to remove the helper if it's not used. In either case, I would probably recommend do this in a custom way in your custom syntax.

Reply:
I have reverted the core change and put improved printing in `AttributesOp`